src/dataset.ts

```diff
@@ -341,6 +347,16 @@ export function checkStructure(s: JsObject): string {
         }
     }
 
+    if ('shape' in s) {
+        const array = s['shape'];
+        if (!Array.isArray(array)) {
+            return `"shape" must be an array`;
+        }
+        if (s.size > 0 && array.length !== s.size) {
+            return `wrong size for "shape", expected ${s.size}, got ${array.length}`;
```

Comment:
this should also validate the fields in the shape array

Reply:
How so?

Reply:
checking that there is a field call `kind` for each element in the array, which is a string; then if it is "sphere", validate the corresponding fields, and similarly for "ellispoid"

Reply:
Ah, I had this originally, but it caused some ugliness with imports. How critical is the type check here? For all we know they may have shapes stored but no intention to use them

Reply:
Added!

Reply:
Put in a rather annoying-to-write type check